x-pack/plugins/ml/public/application/components/data_grid/common.ts

```diff
@@ -158,6 +160,50 @@ export const getDataGridSchemaFromKibanaFieldType = (
   return schema;
 };
 
+/**
+ * Helper to transform feature importance flattened fields with arrays back to object structure
+ *
+ * @param row - EUI data grid data row
+ * @param mlResultsField - Data frame analytics results field
+ * @returns nested object structure of feature importance values
+ */
+export const getFeatureImportance = (row: any, mlResultsField: string): FeatureImportance[] => {
+  const featureNames: string[] = row[`${mlResultsField}.feature_importance.feature_name`];
+  const classNames: string[] = row[`${mlResultsField}.feature_importance.classes.class_name`];
+  const classImportance: number[] = row[`${mlResultsField}.feature_importance.classes.importance`];
+
+  return featureNames.map((fName, index) => {
+    const offset = featureNames.length * index;
+    const featureClassNames = classNames.slice(offset, offset + featureNames.length);
```

Comment:
This is causing the page to crash with regression jobs:

![image](https://user-images.githubusercontent.com/7405507/97586925-bdaddc00-19f2-11eb-9deb-df034a85549a.png)

Reply:
I think this is because the expected shape coming from regression job is different compared to result from classification job here.
<img width="731" alt="Screen Shot 2020-10-29 at 09 28 53" src="https://user-images.githubusercontent.com/43350163/97587393-57f92a00-19c9-11eb-9a65-c66f83fbe6d5.png">

Reply:
Confirmed this is fixed by https://github.com/elastic/kibana/pull/82011/commits/96ccaef1be6614f3dce368ea228c63282fdf6327